### PR TITLE
SAVAS-Lesson 17 was finished.

### PR DESCRIPTION
**Pull request by savaspamuk**

### CODE REVIEW

File: JS/Lesson17/savas/script.js [hunk 1]
```diff
@@ -2,97 +2,233 @@
...
+}
+
+console.log("Exercise 7: ", objectKeys({name: "John", age: 30, city: "New York"}));
+
 console.log('----------------------------');
 // 8. Write a function that takes an object and returns the values as an array.
 // Input: {name: "John", age: 30, city: "New York"}
 // Output: ["John", 30, "New York"]
 
+function objectValues(givenObjects) {
```
Comment (annasvst) [inline JS/Lesson17/savas/script.js:118]:
```suggestion
function getObjectValues(object) {
```

File: JS/Lesson17/savas/script.js [hunk 2]
```diff
@@ -2,97 +2,233 @@
...
 console.log('----------------------------');
 // 9. Write a function that takes an object and a key, and returns true if the key exists in the object, otherwise false.
 // Input: ({name: "John", age: 30, city: "New York"}, "age")
 // Output: true
 // Input: ({name: "John", age: 30, city: "New York"}, "surname")
 // Output: false
 
+function isKeyHere(givenObjects, givenKeys) {
+    if (typeof givenObjects !== "object") {
+      console.log("Provide value is not an object");
```
Comment (annasvst) [inline JS/Lesson17/savas/script.js:133]:
```suggestion
      console.log("Provided value is not an object");
```

File: JS/Lesson17/savas/script.js [hunk 3]
```diff
@@ -2,97 +2,233 @@
...
+console.log("Exercise 8: ", objectValues({name: "John", age: 30, city: "New York"}));
+
 console.log('----------------------------');
 // 9. Write a function that takes an object and a key, and returns true if the key exists in the object, otherwise false.
 // Input: ({name: "John", age: 30, city: "New York"}, "age")
 // Output: true
 // Input: ({name: "John", age: 30, city: "New York"}, "surname")
 // Output: false
 
+function isKeyHere(givenObjects, givenKeys) {
```
Comment (annasvst) [inline JS/Lesson17/savas/script.js:131]:
```suggestion
function isKeyInObject(object, testKey) {
```

File: JS/Lesson17/savas/script.js [hunk 4]
```diff
@@ -2,97 +2,233 @@
...
+    if (typeof givenObjects !== "object") {
+      console.log("Provide value is not an object");
+    } else {
+      if (givenKeys in givenObjects) {
```
Comment (annasvst) [inline JS/Lesson17/savas/script.js:135]: There's an easier way to check it: `object.hasOwnProperty(testKey);`

File: JS/Lesson17/savas/script.js [hunk 5]
```diff
@@ -2,97 +2,233 @@
...
+console.log("Exercise 10: ", addKeyValue({name: "John", age: 30}, "city", "New York"));
+
 console.log('----------------------------');
 // 11. Write a function that takes an object and a key, and removes the key-value pair from the object.
 // Input: ({name: "John", age: 30, city: "New York"}, "city")
 // Output: {name: "John", age: 30}
 
+function removeKeyValue(object, key) {
+  delete object[key];
+  return object
```
Comment (annasvst) [inline JS/Lesson17/savas/script.js:165]:
```suggestion
  return object;
```

**Review by annasvst — COMMENTED** [review]
Very well done Savaş 👏

File: JS/Lesson17/savas/script.js [hunk 6]
```diff
@@ -2,97 +2,233 @@
...
+
+console.log("Exercise 5: ", findEvenNumbers(arrayEx));
+
 console.log('----------------------------');
 // 6. Write a function that takes an array and a value, and returns a new array with all occurrences of the value removed.
 // Input: ([1, 2, 3, 4, 2], 2)
 // Output: [1, 3, 4]
 
+function removeNumber(num, removeNum) {
+  let remainArray = [];
```
[the remark this answers is not in the document]
Comment (savaspamuk) [reply JS/Lesson17/savas/script.js:91]: Sure! Greater than mine :)

File: JS/Lesson17/savas/script.js (hunk 4 above)
Comment (savaspamuk) [reply JS/Lesson17/savas/script.js:135]: Is that ok for now?

**Review by annasvst — APPROVED** [review]
Approved!

**Pull request merged**